tools/Az.Tools.Predictor/Az.Tools.Predictor/Commands/OpenAzSurvey.cs

```diff
@@ -0,0 +1,82 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.PowerShell.Tools.AzPredictor.Utilities;
+using System;
+using System.Diagnostics;
+using System.Globalization;
+using System.IO;
+using System.Management.Automation;
+using System.Text;
+using System.Text.Json;
+
+namespace Microsoft.Azure.PowerShell.Tools.AzPredictor
+{
+    /// <summary>
+    /// <para type="synopsis">Cmdlet to open a survey link in the default browser</para>
+    /// <para type="description">This cmdlet will open a survey link in the default browser. All data from this survey will be anonymized. See the Microsoft Privacy Policy (https://privacy.microsoft.com/) for more information </para>
+    /// </summary>
+    [Cmdlet("Open", "AzSurvey"), OutputType(typeof(bool))]
+    public sealed class OpenAzSurvey : PSCmdlet
+    {
+        private const string _SurveyLinkFormat = "https://aka.ms/azpredictorisurvey?SessionId={0}&Q_CHL=cmdlet";
+        /// <summary>
+        /// <para type="description">Indicates whether the user would like to receive output. </para>
+        /// </summary>
+        [Parameter(Mandatory = false)]
+        public SwitchParameter PassThru { get; set; }
+
+        /// <inheritdoc/>
+        protected override void ProcessRecord()
+        {
+            var profileSettings = Settings.GetProfileSettings();
+            var surveyId = profileSettings?.SurveyId?.ToString(CultureInfo.InvariantCulture) ?? "000000";
+
+            var link = string.Format(OpenAzSurveyLink._SurveyLinkFormat, surveyId, CultureInfo.InvariantCulture);
+
+            Console.WriteLine($"Opening survey {link}");
+
+            // TODO [mahuang] Ouput the link for user to copy/paste in case it's not open in the default browser.
+            var processStartInfo = new ProcessStartInfo();
+            processStartInfo.FileName = link;
+            processStartInfo.UseShellExecute = true;
+
+            Process.Start(processStartInfo);
+
+            if (profileSettings != null)
+            {
+                profileSettings.SurveyId = null;
+
+                try
+                {
+                    var fileContent = JsonSerializer.Serialize<Settings>(profileSettings, new JsonSerializerOptions(JsonUtilities.DefaultSerializerOptions)
+                        {
+                            WriteIndented = true,
+                        });
+                    var profileSettingFilePath = Settings.GetProfileSettingsFilePath();
+                    File.WriteAllText(profileSettingFilePath, fileContent, Encoding.UTF8);
+                }
+                catch
+                {
+                    // Ignore all exceptions.
```

Comment:
@kceiw / @VeryEarly  Should we display here a message with the link if we cannot open the default browser?
For example: "The default browser could not be opened, please copy the following link to a browser to give us yourfeedback"

Reply:
Just chatted with @kceiw  ignore this comment, displaying the survey link on the output of the cmdlet is good.